### DISCUSSION TITLE
Sublime Text 4 Plugin

### DISCUSSION THREAD

**Discussion by biozz** [go-task/task#530] [Show and tell]
Hi,

I am developing [Sublime Text 4 plugin for Taskfile](https://github.com/biozz/sublime-taskfile). 

At the moment, there are these two features:

- Run tasks from the chosen Taskfile (if there is more than one directory with Taskfiles)
- Taskfile initialization in one of the project directories

It displays the output of the tasks in the Quick Panel on the bottom of the editor.

I was able to publish it on packagecontrol.io, it is available [here](https://packagecontrol.io/packages/Taskfile) and you can install it through Package Control.

The only idea for improvement right now is to add Taskfile syntax, so ST can suggest keywords (like it is already implemented in VSCode plugin).

If you are one of those geeks like me, who uses ST and Taskfile and would like to participate in the development of the plugin, you are welcome to join me.

PS: Taskfile maintainers, I don't mind adding this mention in Community section of the docs, I can submit a PR, if you like

**Comment by andreynering** [comment]
Hi @biozz, thanks for posting and creating this plugin!

Yes, please submit a PR to mention this plugin 🙂 

---

Regarding auto completion, as the documentation says, there's both a JSON schema published on the[ official schema store](https://json.schemastore.org/taskfile.json) and another one available by the [VSCode extension](https://github.com/paulvarache/vscode-taskfile/blob/master/schemas/taskfile.schema.json).

On VSCode (and other editors), schemas from schema store are used automatically by default, even without any extension installed. Not sure if Sublime does that as I don't use it in a long time...

**Reply by biozz** [reply]
Oh, thank you for the hint on the json schema! I also didn't know that it can be used in ST. I was actually referring to tm-syntax, which is still widely used in ST's ecosystem. Those tm files are a pain to write and maintain.

Today I found that ST supports json schema with the help of [LSP](https://github.com/sublimelsp/LSP) and [LSP-yaml](https://github.com/sublimelsp/LSP-yaml) plugins which automatically load the schema from  http://schemastore.org/. I'll add a note to the readme of my plugin.
